cloudinit/net/netplan.py

```diff
@@ -263,6 +263,12 @@ def render_network_state(
             header += "\n"
 
         mode = 0o600 if features.NETPLAN_CONFIG_ROOT_READ_ONLY else 0o644
+        if os.path.exists(fpnplan):
```

Comment:
Let's hold off on placing more functionality into util until we have other callers. util is already a very sprawling module at more than 3000 lines which adds an element of time cost/memory cost for loading irrelevant function definitions for most consuming use-cases.  

That said, maybe when we do have other callers which need to reuse the  type of behavior which preserves mode if "at least as strict", we can tackle generalizing this in a util or atomic_helper function. 

Where I can see this being defined generally is an extension or change in behavior of `util.write_file` or `atomic_helper.write_file` to either grow a new param to scope `preserve_if_more_strict` or maybe even change default behavior of `preserve_mode` to only preserve if at least as strict as the request.  Because that will involve a potential change in behavior broadly for write_file

I'll leave that discussion to a separate PR as that could represent change in behavior which affects cloudinit.stages and ssh_util modules I definitely want to defer that effort to a separate PR when we deem it necessary.